platform/src/components/aws/bucket.ts

```diff
@@ -854,6 +917,33 @@ export class Bucket extends Component implements Link.Linkable {
       });
     }
 
+    function createLifecycle() {
+      return output(args.lifecycle).apply((lifecycleRules) => {
+        if (!lifecycleRules || lifecycleRules.length === 0) return;
+
+        return new s3.BucketLifecycleConfigurationV2(
+          ...transform(
+            args.transform?.lifecycle,
+            `${name}Lifecycle`,
+            {
+              bucket: bucket.bucket,
+              rules: lifecycleRules.map((rule, index) => ({
+                id: `${name}LifecycleRule${index}`,
+                status: rule.enabled !== false ? "Enabled" : "Disabled",
+                expiration: rule.expiresIn
+                  ? { days: toSeconds(rule.expiresIn) / 86400 }
+                  : rule.expiresAt
+                    ? { date: rule.expiresAt }
+                    : undefined,
```

Comment:
if it's possible to provide both properties `days` and `date` we should do it, otherwise mention it in the docs